src/gui/src/heatMap.cpp

```diff
@@ -349,15 +349,15 @@ HeatMapDataSource::MapView HeatMapDataSource::getMapView(
   const int x_high = std::min(
       static_cast<int>(
           std::ceil(bounds.xMax() / (getGridXSize() * dbu_per_micron))),
-      static_cast<int>(map_.shape()[0]) - 1);
+      static_cast<int>(map_.shape()[0]));
```

Comment:
wouldn't this be out of bounds? ie. is the array is 10 wide the indexes go from 0 -> 9?

Reply:
Now, I got a little confused. I guess it makes sense, but:

https://github.com/The-OpenROAD-Project/OpenROAD/blob/2f2fd3dbe7d377126468b6f79bd7407da81afc6c/src/gui/src/heatMap.cpp#L631-L635

Here, the drawing iterations are not getting to the last raw/column.

In your example (considering only x coordinate): array is 10 wide,
x_size must be 10 so that the last element 9 is covered, but, as it currently is, x_size=9.

I am missing something?

Reply:
is `::index_range` inclusive or exclusive of the endpoint?

Reply:
Although this is already merged, just FYI: https://www.boost.org/doc/libs/1_82_0/libs/multi_array/doc/reference.html#generators

As described in Associated Types for Views section, it looks like the interval is open in the upper bound.

`::index_range(lower_bound, upper_bound)`
lower_bounds is inclusive, upper_bound is exclusive